data-prepper-plugins/opensearch/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java

```diff
@@ -584,6 +585,34 @@ public void testOpenSearchRoutingField(final String testRoutingField) throws IOE
     sink.shutdown();
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"info/ids/id", "id"})
+  public void testOpenSearchDynamicIndex(final String testIndex) throws IOException, InterruptedException {
```

Comment:
We should also have an integration test that combines a dynamic index based on properties (as you have here) with a time-based index names (e.g. `logs-%{yyyy.MM}`.

Relatedly, I don't see an integration test which actually verifies time-based index names. We should have a test for that in isolation to ensure it is not affected by this change.